fix: Fixes editable table cell paddings for certain corner cases

pan-kot opened the pull request:
### Description

The PR includes two small fixes and minor refactoring for editable table cell styles:
1. Improved editor vertical alignment;
2. Correct editable cell padding in sticky column.

Before:

<img width="1659" alt="Screenshot 2024-12-17 at 10 51 21" src="https://github.com/user-attachments/assets/c8ea6360-853a-408b-8fe2-e28a1c293f22" />

https://github.com/user-attachments/assets/cc480f20-3953-44e6-81c1-53d17d9cda01

After:

<img width="1659" alt="Screenshot 2024-12-17 at 10 52 00" src="https://github.com/user-attachments/assets/c43ea22c-0f99-4071-bf8c-23537e05a26f" />

https://github.com/user-attachments/assets/57ccf36c-2152-432e-be90-04ca496532bc

### How has this been tested?

* Manual testing on "cell-permutations" and "editable" test pages.
* Screenshot tests.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -147,28 +148,21 @@ $cell-negative-space-vertical: 2px;
   }
   &:first-child {
     border-inline-start: $border-placeholder;
+    @include cell-padding-inline-start($cell-edge-horizontal-padding);
```
Moved from `first-child:not(.is-visual-refresh)`. There is no reason to include the ":not(.is-visual-refresh)" because first column padding for VR is overridden anyways.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -147,28 +148,21 @@ $cell-negative-space-vertical: 2px;
   }
+  // Using very small padding for 1st column cells in VR.
   &.is-visual-refresh:first-child {
-    &:not(.has-striped-rows) {
```
Removed "&:not(.has-striped-rows)" because there is an explicit override for ".has-striped-rows" down below.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -147,28 +148,21 @@ $cell-negative-space-vertical: 2px;
-    &.sticky-cell-pad-inline-start:not(.has-selection) {
-      @include cell-padding-inline-start($cell-horizontal-padding);
+    @include cell-padding-inline-start(awsui.$space-xxxs);
+    &.body-cell-editable:hover {
```
Removed the ":not(.body-cell-edit-active)" prefix because the active edit cell cannot be hovered anyways. This also fixes the bug when the wrong padding was assigned for sticky + editable + hovered cells.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -246,9 +242,6 @@ $cell-negative-space-vertical: 2px;
       transition-timing-function: awsui.$motion-easing-sticky;
     }
-    &-pad-left:not(.has-selection):not(.is-visual-refresh.body-cell:first-child.has-striped-rows) {
-      @include cell-padding-inline-start(awsui.$space-table-horizontal);
```
Removed this completely. The table cells should use normal paddings also when sticky.

pan-kot commented on src/table/body-cell/styles.scss:
```diff
@@ -418,7 +411,8 @@ $cell-negative-space-vertical: 2px;
       @include cell-padding-inline-end($editing-cell-padding-inline);
-      @include cell-padding-block($editing-cell-padding-block);
+      @include cell-padding-block-start($editing-cell-padding-block);
+      @include cell-padding-block-end(calc($editing-cell-padding-block + 1px));
```
Adding a small extra space here ensures better input alignment especially in compact mode.

jperals commented:
The inline edit field extends towards the inline-start by ~1px when hovered. This didn't happen before (see [before](https://d21d5uik3ws71m.cloudfront.net/components/12dbf6bfa7fe1984d27d546967321cb442be3b48/dev-pages/index.html#/light/table/inline-editor.permutations) vs [after](https://d21d5uik3ws71m.cloudfront.net/components/b4a3b82bc8207acb9304178455ba6584cec86939/dev-pages/index.html#/light/table/inline-editor.permutations))

https://github.com/user-attachments/assets/de630af0-9e44-4dac-8005-7e8c77ecbdec

pan-kot commented:
> The inline edit field extends towards the inline-start by ~1px when hovered. This didn't happen before (see [before](https://d21d5uik3ws71m.cloudfront.net/components/12dbf6bfa7fe1984d27d546967321cb442be3b48/dev-pages/index.html#/light/table/inline-editor.permutations) vs [after](https://d21d5uik3ws71m.cloudfront.net/components/b4a3b82bc8207acb9304178455ba6584cec86939/dev-pages/index.html#/light/table/inline-editor.permutations))
> 
>  Untitled.mov

Good finding! Somehow that wasn't observed in the table itself.

I updated styles so that cell hover override is applied for multiple conditions taking the corresponding padding. That also fixed slight cell shaking in the first column in certain conditions.

Before (main):

https://github.com/user-attachments/assets/00eca84a-51f9-406c-9900-8c4d8d14a26a

After:

https://github.com/user-attachments/assets/51c2c36c-07c8-4457-8916-c08b76dabd00
